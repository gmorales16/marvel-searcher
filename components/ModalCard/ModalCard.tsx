import React from "react";
import styled from "styled-components";
import { CiStar } from "react-icons/ci";

const Card = styled.div`
  display: flex;
  max-width: 600px;
  margin: 10px;
  padding: 10px;
`;

const CardImage = styled.img`
  flex: 0 0 200px; /* Establecemos un ancho fijo para la imagen */
  height: 200px;
  object-fit: cover;
  border-radius: 4px;
  margin-right: 10px; /* Agregamos un margen a la derecha para separar la imagen */
`;

const CardContent = styled.div`
  display: flex;
  flex-direction: column;
  flex: 1; /* El contenido ocuparÃ¡ el espacio disponible restante */
`;

const CardHeader = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
`;

const CardTitle = styled.h3`
  margin: 0;
  color: #505050;
`;

const CardDescription = styled.p`
  margin-top: 10px;
  color: #505050;
  white-space: pre-wrap;
`;

const StarIcon = styled(CiStar)`
  color: #505050;
  font-size: 24px;
`;

interface ModalCardProps {
  url: string;
  title: string;
  description: string;
  id: Number;
}

export default function ModalCard({
  url,
  title,
  description,
  id,
}: ModalCardProps) {
  // Set ID Comic in LocalStorage

  const handleClickCard = () => {
    localStorage.setItem("idComic", String(id));
    window.location.href = `/comic`;
  };
  return (
    <Card onClick={handleClickCard}>
      <CardImage src={url} alt="Card Image" />
      <CardContent>
        <CardHeader>
          <CardTitle>{title}</CardTitle>
          <StarIcon />
        </CardHeader>
        <CardDescription>{description}</CardDescription>
      </CardContent>
    </Card>
  );
}
