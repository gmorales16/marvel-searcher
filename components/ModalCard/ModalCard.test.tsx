import "jsdom-global/register";
import "@testing-library/jest-dom/extend-expect";
import React from "react";
import { render, screen } from "@testing-library/react";
import ModalCard from "./ModalCard";

// Mock the react-icons library
jest.mock("react-icons/ci", () => ({
  CiStar: () => <div data-testid="star-icon" />,
}));

describe("ModalCard", () => {
  test("renders with correct title and description", () => {
    const mockProps = {
      url: "Hi im mock-url",
      title: "Leon Scott Kennedy",
      description: "Mock Description",
      id: 1,
    };
    render(<ModalCard {...mockProps} />);

    const titleElement = screen.getByText("Leon Scott Kennedy");
    const descriptionElement = screen.getByText("Mock Description");
    const iconElement = screen.getByTestId("star-icon");

    expect(titleElement).toBeInTheDocument();
    expect(descriptionElement).toBeInTheDocument();
    expect(iconElement).toBeInTheDocument();
    const { asFragment } = render(<ModalCard {...mockProps} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
