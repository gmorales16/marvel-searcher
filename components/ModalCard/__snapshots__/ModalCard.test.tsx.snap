// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalCard renders with correct title and description 1`] = `
<DocumentFragment>
  <div
    class="ModalCard__Card-sc-mkd39d-0 eDWtCQ"
  >
    <img
      alt="Card Image"
      class="ModalCard__CardImage-sc-mkd39d-1 eXNKxM"
      src="Hi im mock-url"
    />
    <div
      class="ModalCard__CardContent-sc-mkd39d-2 DKSmK"
    >
      <div
        class="ModalCard__CardHeader-sc-mkd39d-3 fZVkzu"
      >
        <h3
          class="ModalCard__CardTitle-sc-mkd39d-4 fFAfj"
        >
          Leon Scott Kennedy
        </h3>
        <div
          data-testid="star-icon"
        />
      </div>
      <p
        class="ModalCard__CardDescription-sc-mkd39d-5 bpsHBy"
      >
        Mock Description
      </p>
    </div>
  </div>
</DocumentFragment>
`;
